spec/bolt/transport/local_spec.rb

```diff
@@ -0,0 +1,246 @@
+require 'spec_helper'
+require 'bolt_spec/errors'
+require 'bolt_spec/files'
+require 'bolt_spec/task'
+require 'bolt/transport/local'
+require 'bolt/config'
+
+describe Bolt::Transport::Local do
+  include BoltSpec::Errors
+  include BoltSpec::Files
+  include BoltSpec::Task
+
+  let(:local) { Bolt::Transport::Local.new }
+  let(:echo_script) { <<BASH }
+for var in "$@"
+do
+    echo $var
+done
+BASH
+
+  let(:target) { Bolt::Target.new('local://localhost') }
+
+  def result_value(stdout = nil, stderr = nil, exit_code = 0)
+    { 'stdout' => stdout || '',
+      'stderr' => stderr || '',
+      'exit_code' => exit_code }
+  end
+
+  context "when executing", bash: true do
+    it "executes a command" do
+      expect(local.run_command(target, 'echo $USER').value['stdout'].strip).to eq(ENV['USER'])
```

Comment:
I'd suggest using the more widely supported `HOME` environment variable rather than the `USER`.